platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -47,6 +47,46 @@ Note that certain features that rely on third-party tools, such as [PMD](https:/
 
 For details, see the full [compatibility documentation](userguide/compatibility.html#java).
 
+<a name="java-compiler-error-rendering"></a>
+### Java compiler errors in the failure report
+
+From now on, Gradle also includes Java compilation failures in the report at the end of the build.
```

Comment:
I think we should start with the context. What's the problem we wanted to solve:
- in previous Gradle versions, the build solely relied on the compiler output
- If build failed, the message just had an advice to find it in the build log
- For large builds and for CI this means tedious scrolling finding the corresponding compile task output

Reply:
FYI, I've wanted to tell this story, but I was unsure how much storytelling we want to do in the release notes. I guess by your comment, you also think it would be worthwhile, so I will expand upon the details

Reply:
I've added @lkasso to this review explicitly to help us with the content.

Reply:
Pushed the new text, then resolve or comment

Reply:
Left CI story out, my bad

Reply:
Though new text might be enough, and we don't need to shoehorn in CI there.